utilities/transactions/lock/range/range_tree/range_tree_lock_tracker.cc

```diff
@@ -0,0 +1,65 @@
+//
+// A lock tracker to be used together with RangeTreeLockManager
+//
+#ifndef ROCKSDB_LITE
+#ifndef OS_WIN
+
+#include "range_tree_lock_tracker.h"
+
+#include "range_tree_lock_manager.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+RangeLockList *RangeTreeLockTracker::getOrCreateList() {
+  RangeLockList *res;
+  if ((res = getList())) return res;
+
+  // Doesn't exist, create
+  range_list.reset(new RangeLockList());
+  return getList();
+}
+
+void RangeTreeLockTracker::Track(const PointLockRequest &lock_req) {
+  DBT key_dbt;
+  std::string key;
+  serialize_endpoint(Endpoint(lock_req.key, false), &key);
+  toku_fill_dbt(&key_dbt, key.data(), key.size());
+  RangeLockList *rl = getOrCreateList();
+  rl->append(lock_req.column_family_id, &key_dbt, &key_dbt);
+}
+
+void RangeTreeLockTracker::Track(const RangeLockRequest &lock_req) {
+  DBT start_dbt, end_dbt;
+  std::string start_key, end_key;
+
+  serialize_endpoint(lock_req.start_endp, &start_key);
+  serialize_endpoint(lock_req.end_endp, &end_key);
+
+  toku_fill_dbt(&start_dbt, start_key.data(), start_key.size());
+  toku_fill_dbt(&end_dbt, end_key.data(), end_key.size());
+
+  RangeLockList *rl = getOrCreateList();
+  rl->append(lock_req.column_family_id, &start_dbt, &end_dbt);
+}
+
+PointLockStatus RangeTreeLockTracker::GetPointLockStatus(
+    ColumnFamilyId /*cf_id*/, const std::string & /*key*/) const {
+  // TODO: what to do here if we are holding a range lock that is embedding the
```

Comment:
If there is a range lock containing the key, then from point lock's perspective, the point is locked?

Reply:
It is locked, in a sense that an attempt to get the lock on a key will fail.
But it is not "tracked", in a sense that we haven't made a note of what was its SequenceNumber when the range lock was acquired.

I don't think this comment is relevant as `class RangeTreeLockTracker` has  `RangeTreeLockTracker::IsPointLockSupported()` which returns FALSE, so there shouldn't be anybody calling this method.

(and providing an implementation that checks if there's a lock on a range that contains this point - this is going to be hard as the RangeLockList that stores the locks we've acquired is implemented as unsorted array, which doesn't allow lookups).

I think I'll just modify the comment to say that this function is not expected to be implemented.